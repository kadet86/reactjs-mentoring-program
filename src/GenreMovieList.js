import React from 'react';
import MovieList from './MovieList';
import {Toolbar} from 'primereact/toolbar';

const URL = 'https://reactjs-cdp.herokuapp.com/movies';

export default class GenreMovieList extends React.PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            movies: []
        };
    }

    fetchMovies() {
        fetch(`${URL}?sortOrder=desc&limit=6&sortBy=vote_average&searchBy=genre&search=${this.props.genre}`)
            .then(res => res.json())
            .then(json => {
                this.setState({movies: json.data});
            });
    }

    componentDidMount() {
        this.fetchMovies();
    }

    render() {
        return (
            <section>
                <Toolbar className="movie-list-toolbar">
                    <label className="movie-list-toolbar__label">Films by {this.props.genre} genre</label>
                </Toolbar>
                <MovieList 
                    movies={this.state.movies} 
                    navigateToMovie={this.props.navigateToMovie} />
            </section>
        );
    }
}