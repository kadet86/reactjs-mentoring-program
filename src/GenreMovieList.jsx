import React from 'react';
import MovieList from './MovieList';
import {Toolbar} from 'primereact/toolbar';
import {connect} from 'react-redux';
import {getMovies} from './actions';

class GenreMovieList extends React.PureComponent {
    fetchMovies() {
        this.props.getMovies({
            limit: 9,
            sortBy: 'vote_average',
            searchBy: 'genres',
            query: this.props.genre,
        });

        document.documentElement.scrollTop = 0;
    }

    componentDidMount() {
        this.fetchMovies();
    }

    componentDidUpdate(prevProps) {
        if (this.props.genre !== prevProps.genre) {
            this.fetchMovies();
        }
    }

    render() {
        return (
            <section>
                <Toolbar className="movie-list-toolbar">
                    <label className="movie-list-toolbar__label">Films by {this.props.genre} genre</label>
                </Toolbar>
                <MovieList 
                    movies={this.props.movies} 
                    navigateToMovie={this.props.navigateToMovie} />
            </section>
        );
    }
}

const mapStateToProps = (state) => ({
    movies: state.movies,
});

const mapDispatchToProps = {getMovies};

export default connect(mapStateToProps, mapDispatchToProps)(GenreMovieList);
