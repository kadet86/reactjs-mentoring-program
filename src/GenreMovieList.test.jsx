import React from 'react';
import {shallow} from 'enzyme';
import GenreMovieList from './GenreMovieList';

describe('GenreMovieList component', () => {
    beforeEach(() => {
        fetch.resetMocks();
    });

    it('has persistent default snapshot', () => {
        fetch.mockResponseOnce(JSON.stringify({ data: [] }));
        const component = shallow(<GenreMovieList />);
        expect(component.html()).toMatchSnapshot();
    });

    it('fetches movies when genre is changed', () => {
        fetch.mockResponse(JSON.stringify({ data: [] }));
        const component = shallow(<GenreMovieList />);
         
        component.setProps({genre: 'action'});

        expect(fetch.mock.calls[1][0]).toEqual('https://reactjs-cdp.herokuapp.com/movies?sortOrder=desc&limit=6&sortBy=vote_average&searchBy=genres&search=action');
    });
});
