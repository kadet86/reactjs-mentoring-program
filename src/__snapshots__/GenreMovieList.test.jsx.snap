// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreMovieList component has persistent default snapshot 1`] = `"<section><div class=\\"p-toolbar p-component movie-list-toolbar\\"><label class=\\"movie-list-toolbar__label\\">Films by  genre</label></div><section class=\\"movie-list\\"></section></section>"`;
